pkg/buildcontext/git.go

```diff
@@ -41,6 +41,28 @@ func (g *Git) UnpackTarFromBuildContext() (string, error) {
 	if len(parts) > 1 {
 		options.ReferenceName = plumbing.ReferenceName(parts[1])
 	}
-	_, err := git.PlainClone(directory, false, &options)
+	r, err := git.PlainClone(directory, false, &options)
+
+	if err == nil && len(parts) > 2 {
+		// ... retrieving the commit being pointed by HEAD
+		_, err := r.Head()
```

Comment:
Can you please add more explanation around why this is necessary?
It would be great to add some tests around this integration. 
Please see your existing Git Context integration test.